relay-general/src/protocol/measurements.rs

```diff
@@ -0,0 +1,207 @@
+use crate::types::{Annotated, Error, FromValue, Object, Value};
+
+/// An individual observed measurement.
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, ToValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct Measurement {
+    /// Value of observed measurement value.
+    #[metastructure(required = "true")]
+    pub value: Annotated<f64>,
+}
+
+/// A map of observed measurement values.
+///
+/// Measurements are only available on transactions. They contain measurement values of observed
+/// values such as Largest Contentful Paint (LCP).
+#[derive(Clone, Debug, Default, PartialEq, Empty, ToValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct Measurements(pub Object<Measurement>);
+
+impl FromValue for Measurements {
+    fn from_value(value: Annotated<Value>) -> Annotated<Self> {
+        let mut processing_errors = Vec::new();
+
+        let mut measurements = Object::<Measurement>::from_value(value).map_value(|measurements| {
+            let measurements = measurements
+                .into_iter()
+                .filter_map(|(name, value)| {
+                    let name = name.trim();
+                    if let Some(measurement_value) = value.value() {
+                        if is_valid_measurement_name(name) {
+                            if measurement_value.value.value().is_some() {
+                                return Some((name.to_lowercase(), value));
+                            } else {
+                                let new_value = Measurement {
+                                    value: Annotated::from_error(
+                                        Error::invalid("measurement value must be numeric"),
+                                        None,
+                                    ),
+                                };
+
+                                return Some((name.to_lowercase(), Annotated::new(new_value)));
```

Comment:
Attaching the processing error here instead of at the map level will look like this:

<img width="1337" alt="Screen Shot 2020-09-24 at 8 44 59 AM" src="https://user-images.githubusercontent.com/139499/94148172-46f64f80-fe44-11ea-9c86-66bd6fa140b2.png">

Reply:
Why can't we have the error in-place for the remaining measurements as well? I'd say that's what we should do and have downstream deal with `null` in those places (like we already have to with the inner value field)

Reply:
@untitaker I don't quite follow. If the remaining measurements are valid, they won't have errors.

Reply:
You are currently filtering out invalid measurements vs replacing them with `null`, I think we should go with setting them to `null` like you showed in teh screenshot basically.

Reply:
@dashed I think this is the only thing I am rather adamant about, the rest seems fine and I'd merge cc @jan-auer

Reply:
@untitaker They're now replaced with `null`s. I've already included some tests to cover these test cases. 

Unless there's a test case I missed 🤔

Reply:
Ah sorry you're doing this only for invalid key names, nvm